src/A2A/JsonRpc/JsonRpcRequestConverter.cs

```diff
@@ -0,0 +1,160 @@
+            using var jsonDoc = JsonDocument.ParseValue(ref reader);
+            var rootElement = jsonDoc.RootElement;
+
+            // Validate the JSON-RPC request structure
+            requestId = ReadAndValidateIdField(rootElement);
+
+            return new JsonRpcRequest
+            {
+                Id = requestId,
+                JsonRpc = ReadAndValidateJsonRpcField(rootElement, requestId),
+                Method = ReadAndValidateMethodField(rootElement, requestId),
+                Params = ReadAndValidateParamsField(rootElement, requestId)
+            };
+        }
+        catch (JsonException ex)
+        {
+            throw new A2AException("Invalid JSON-RPC request payload.", ex, A2AErrorCode.ParseError);
+        }
+    }
+
+    public override void Write(Utf8JsonWriter writer, JsonRpcRequest value, JsonSerializerOptions options)
+    {
+        if (value is null)
+        {
+            throw new ArgumentNullException(nameof(value), "Cannot serialize a null JsonRpcRequest.");
+        }
+
+        writer.WriteStartObject();
+        writer.WriteString("jsonrpc", value.JsonRpc);
+        writer.WriteString("id", value.Id);
+        writer.WriteString("method", value.Method);
+
+        if (value.Params.HasValue)
+        {
+            writer.WritePropertyName("params");
+            value.Params.Value.WriteTo(writer);
+        }
+
+        writer.WriteEndObject();
+    }
+
+    /// <summary>
+    /// Reads and validates the 'id' field of a JSON-RPC request.
+    /// </summary>
+    /// <param name="rootElement">The root JSON element containing the request.</param>
+    /// <returns>The extracted request ID as a string, or null if not present.</returns>
+    private static string? ReadAndValidateIdField(JsonElement rootElement)
+    {
+        if (rootElement.TryGetProperty("id", out var idElement))
+        {
+            if (idElement.ValueKind != JsonValueKind.String &&
+                idElement.ValueKind != JsonValueKind.Number &&
+                idElement.ValueKind != JsonValueKind.Null)
+            {
+                throw new A2AException("Invalid JSON-RPC request: 'id' field must be a string, number, or null.", A2AErrorCode.InvalidRequest);
+            }
+
+            return idElement.ValueKind == JsonValueKind.Null ? null : idElement.ToString();
+        }
+
+        return null;
+    }
+
+    /// <summary>
+    /// Reads and validates the 'jsonrpc' field of a JSON-RPC request.
+    /// </summary>
+    /// <param name="rootElement">The root JSON element containing the request.</param>
+    /// <param name="requestId">The request ID for error context.</param>
+    /// <returns>The JSON-RPC version as a string.</returns>
+    private static string ReadAndValidateJsonRpcField(JsonElement rootElement, string? requestId)
+    {
+        if (rootElement.TryGetProperty("jsonrpc", out var jsonRpcElement))
+        {
+            var jsonRpc = jsonRpcElement.GetString();
+
+            if (jsonRpc != "2.0")
```

Comment:
nit how about:
```suggestion
            if (jsonRpc is not JsonRpcSupportedVersion)
```

and add a constant?